packages/playwright-core/src/server/chromium/crBrowser.ts

```diff
@@ -312,9 +313,28 @@ class CRServiceWorker extends Worker {
     session.once('Runtime.executionContextCreated', event => {
       this._createExecutionContext(new CRExecutionContext(session, event.context));
     });
+
+    // FIXME(raw): This will all be moved into a crServiceWorkerNetworkManager that will use shared bits that page/frame-level network mgr. (currently crNetworkManager) will depend on
```

Comment:
I'd rather reuse CRNetworkManager as is. This way we'll get all the nice logic in there, and will also see where else we should hook up service workers (e.g. `authenticate` or `setOffline`).  What are the blockers for that?

Reply:
The big one is ServiceWorker's won't be intrinsically tied to a `page` (or `frame`) and CRNetworkManager is frame/page-centric as is. The alternative motive was to see which bits (and nuance) of CRNetworkManager are absolutely necessary in the context of SW. Building it from first principles also gives me an opportunity to better understand how/when the nuanced bits of CRNetworkManager become necessary (so in this way it's more of a learning exercise). Although, in the end I may end up throwing away CRServiceWorkerNetworkManager and just taking the learning along! As I get a bit deeper, I may cut my learning exercise short since the convergence looks more obvious than it did before I started! 

@pavelfeldman and I discussed this topic a bit so CC'ing in case he has comments to add!